dff/script/core/script.py

```diff
@@ -25,6 +26,80 @@
 logger = logging.getLogger(__name__)
 
 
+def error_handler(error_msgs: list, msg: str, exception: Optional[Exception] = None):
+    """
+    This function handles errors during :py:class:`~dff.script.Script` validation.
+
+    :param error_msgs: List that contains error messages. :py:func:`~dff.script.error_handler`
+        adds every next error message to that list.
+    :param msg: Error message which is to be added into `error_msgs`.
+    :param exception: Invoked exception. If it has been set, it is used to obtain logging traceback.
+        Defaults to `None`.
+    :param logging_flag: The flag which defines whether logging is necessary. Defaults to `True`.
+    """
+    error_msgs.append(msg)
+    logger.error(msg, exc_info=exception)
+
+
+def validate_callable(callable: Callable, name: str, flow_label: str, node_label: str) -> List:
+    """
+    This function validates a function during :py:class:`~dff.script.Script` validation.
+    It checks parameter number (unconditionally), parameter types (if specified) and return type (if specified).
+
+    :param callable: Function to validate.
+    :param name: Type of the function (label, condition, response, etc.).
+    :param flow_label: Flow label this function is related to (used for error localization only).
+    :param node_label: Node label this function is related to (used for error localization only).
+    :param error_msgs: List that contains error messages. All the error message will be added to that list.
+    :param logging_flag: The flag which defines whether logging is necessary. Defaults to `True`.
+    :param expected_types: Tuple of types that correspond to expected function parameter types.
+        Also used for parameter number check. If `None`, parameter check is skipped. Defaults to `None`.
+    :param return_type: Tuple, containing expected function return type.
+        If `None` or contains more or less than 1 element, return type check is skipped. Defaults to `None`.
+    :return: list of produced error messages.
+    """
+    from dff.script.core.context import Context
+    from dff.pipeline.pipeline.pipeline import Pipeline
+
+    USER_FUNCTION_TYPES = {
```

Comment:
Define these as `Enum` and use pass `Enum` in this function instead of `str`.

Reply:
Another option is to replace this line:
```
- function_type = name if name in USER_FUNCTION_TYPES.keys() else "processing"
+ if name not in USER_FUNCTION_TYPES:
+     raise ...
```
ATM if an incorrect type is passed we just ignore that.